### PR TITLE
LPD-20661 

### PR DESCRIPTION
**Pull request by rijema**
Hi team! Let me know regarding the changes of this test for LPD-20124

Related issue: https://liferay.atlassian.net/browse/LPD-20661

### CODE REVIEW

**Comment by marcelabc** [comment]
Just started reviewing :)

**Comment by marcelabc** [comment]
I just wanted to let you know that I edited the description to include the "Related Issue" (with the Jira ticket). Please always add this information in your following reviews if you can! :smile:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:160]: I saw that the other methods in this class are testing the items values and here you are just asserting the size. Could you assert the values? Why did you decide just to assert the size?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:126]: Could you rename to `testGetInfoFieldValuesFieldSetDDMFormFieldType`? This is to be in line with the names of the other methods.
What do you think?

Tip: sometimes we call "nested fields," but it is best to follow the name defined in the code in the tests. This type of field is named "fieldset."

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:128]: Could you rename this file name to `test-ddm-structure-definition-with-nested-fields.json`? 
What do you think?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:143]: Could you rename this file name to `ddm-structure-fieldset-field.sxd` ? This is to be in line with the names of the other files.
What do you think?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json:15]: Since there are more than one fields group, can you add 1/2/3 in the end?
`"en_US": "Fields Group 1" `

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json:19]: Repeat the same fieldReference here.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json:34]: You can put Text1, Text2...

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json
Comment (marcelabc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json:19]: and to the rest of JSON file please.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json
Comment (marcelabc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json:15]: and to the rest of JSON file please.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json
Comment (marcelabc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json:34]: and to the rest of JSON file please.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:160]: Would be something like this?

for(DDMFormField ddmFormField : ddmStructure.getDDMFormFields(true)){
			if(!ddmFormField.getName().contains("FieldSet")){
				_assertGetInfoFieldValues(
					ddmFormField.getName(), journalArticle,
					Assert::assertNull);
			}
		}
		
		The _assertGetInfoFieldValues doesn't support nested fields for now, so I thought in order to match the expected results of the ticket just to know if all the fields were being returned, i.e. the quantity

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:138]: Yesss, I need it :( I had a ton of problems when using this Test class, as they're not prepared to handle nested fields. I spent some days trying to insert nested fields to use the Journal Test classes, but to be honest the changes was taking too long. So talking with Carolina, she did an amazing suggestion of mocking the data structure, I never thought in doing it.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:130]: Sure!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:271]: Sure!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:404]: Here we have some problems. Since the signature to this package is the same as the one I'm changing in InfoFieldValuesTest class, the code doesn't understand I want to look for a DDMBaseTest in another folder and try to find in the current visible worktree, returning me a non-existing Class. In order to solve this, I extended from the BaseServiceTest class. Could you please take a look?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:284]: image

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:280]: Sure!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:126]: It makes a lot of sense!!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json:15]: Sure!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:128]: Sure!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:143]: Sure!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json:19]: Sure!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-structured-content-nested-structure.json:34]: Ok

**Comment by rijema** [comment]
Hi @marcelabc , I did most part of the changes and updated with a new commit, let me know if this commit message could be this one. The only thing I'm a bit 'cabreiro' /aka suspicious/ is regarding this change 

https://github.com/liferay-forms/liferay-portal/pull/2503#discussion_r1557476396

Let me know how we're going to handle this change. Thanks!

**Comment by marcelabc** [comment]
Just started reviewing :)

**Comment by marcelabc** [comment]
Hey @rijema 
Whenever you finish applying the review changes, can you please remove the flag and ask for another review?
If you don't have permission to edit the labels, let me know :smile:

**Comment by marcelabc** [comment]
Regarding this comment:
> I did most part of the changes and updated with a new commit, let me know if this commit message could be this one.

The reviewer should review the code in the final version. 
I believe this commit should be squashed with the one that introduced the change. An extra commit SF is just confusing for me. I prefer to review the code/logic in the final version. 

Since the reviewer can review the PR per commit, a change in the same class but in different commits can slow down the review process because the reviewer will stop and write a comment, and this suggestion is implemented in X commits later.

If you don't want to refactor it now, you can leave it as it is.
However, I would like you to keep this tip for the future PRs.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:155]: You can replace this variable for `RandomTestUtil.randomLocaleStringMap()` and inline where you are using.

Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:155]: Okay.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (marcelabc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:168]: Yeah, it can be something like this.
If you use ddmStructure.getDDMFormFields(**true**), I believe the fieldset type will not be included in that list. This happens because a transient field is a field that does not have value (for example, fieldset, separator).

**Comment by rijema** [comment]
Hi @marcelabc , changes done =)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java]: I believe you're saying false, is that right? Really good to know this!! Thank you for the amazing tip.

**Comment by marcelabc** [comment]
Hey @rijema 
Whenever you push your changes, remove the `Author Action Required` label and ask for another review please :smile:

**Comment by rijema** [comment]
Ok @marcelabc , thank you for letting me know!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/info/item/provider/test/DDMFormValuesInfoFieldValuesProviderTest.java:319]: Can you remove the parameter size and use `ddmFormFields.size()` instead?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-ddm-structure-definition-with-nested-fields.json
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-ddm-structure-definition-with-nested-fields.json:38]: Can you edit it to "Text 1," please?🥹
Do this to the other's Text Label.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. It was a minor change. Thank you!!
Let me know if you have questions :smile:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-ddm-structure-definition-with-nested-fields.json
Comment (rijema) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/resources/com/liferay/dynamic/data/mapping/dependencies/test-ddm-structure-definition-with-nested-fields.json]: Of course! Done. Thx!!

**Comment by rijema** [comment]
ci:test:sf

**Review by marcelabc — APPROVED** [review]
LGTM 🎉
I appreciate your patience :)

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
Since we are implementing a new integration test, we don't need to run the forms suite.
The relevant suite runs all the integration tests :+1:

**Comment by rijema** [comment]
Don't worry about time, I learned a lot with your help. Thanks again!

**Comment by rijema** [comment]
Should I submit this PR manually, @marcelabc ?

**Pull request closed without merging**
